TYPO3.Media/Classes/TYPO3/Media/Domain/Model/ThumbailConfiguration.php

```diff
@@ -0,0 +1,137 @@
+ * It is free software; you can redistribute it and/or modify it under    *
+ * the terms of the GNU General Public License, either version 3 of the   *
+ * License, or (at your option) any later version.                        *
+ *                                                                        */
+
+use Doctrine\ORM\Mapping as ORM;
+use TYPO3\Flow\Annotations as Flow;
+
+/**
+ * An Thumbail Configuration
+ */
+class ThumbailConfiguration {
+
+	/**
+	 * @var integer
+	 */
+	protected $width;
+
+	/**
+	 * @var integer
+	 */
+	protected $maximumWidth;
+
+	/**
+	 * @var integer
+	 */
+	protected $height;
+
+	/**
+	 * @var integer
+	 */
+	protected $maximumHeight;
+
+	/**
+	 * @var boolean
+	 */
+	protected $allowCropping;
+
+	/**
+	 * @var boolean
+	 */
+	protected $allowUpScaling;
+
+	/**
+	 * ImageConfiguration constructor.
+	 * @param integer $width
+	 * @param integer $maximumWidth
+	 * @param integer $height
+	 * @param integer $maximumHeight
+	 * @param boolean $allowCropping
+	 * @param boolean $allowUpScaling
+	 */
+	public function __construct($width = NULL, $maximumWidth = NULL, $height = NULL, $maximumHeight = NULL, $allowCropping = FALSE, $allowUpScaling = FALSE) {
+		$this->width = $width;
+		$this->maximumWidth = $maximumWidth;
+		$this->height = $height;
+		$this->maximumHeight = $maximumHeight;
+		$this->allowCropping = $allowCropping;
+		$this->allowUpScaling = $allowUpScaling;
+	}
+
+	/**
+	 * @return integer
+	 */
+	public function getWidth() {
+		if ($this->width !== NULL) {
+			return $this->width;
+		}
+		// @deprecated since 2.0, simulate the behaviour of 1.2
+		if ($this->isCroppingAllowed() && $this->getMaximumWidth() !== NULL && $this->getMaximumHeight() !== NULL) {
+			return $this->getMaximumWidth();
+		}
+		return NULL;
+	}
+
+	/**
+	 * @return integer
+	 */
+	public function getMaximumWidth() {
+		return $this->maximumWidth;
+	}
+
+	/**
+	 * @return integer
+	 */
+	public function getHeight() {
+		if ($this->height !== NULL) {
+			return $this->height;
+		}
+		// @deprecated since 2.0, simulate the behaviour of 1.2
+		if ($this->isCroppingAllowed() && $this->getMaximumWidth() !== NULL && $this->getMaximumHeight() !== NULL) {
+			return $this->getMaximumHeight();
+		}
+		return NULL;
+	}
+
+	/**
+	 * @return integer
+	 */
+	public function getMaximumHeight() {
+		return $this->maximumHeight;
+	}
+
+	/**
+	 * @return boolean
+	 */
+	public function getRatioMode() {
+		return ($this->isCroppingAllowed() ? ImageInterface::RATIOMODE_OUTBOUND : ImageInterface::RATIOMODE_INSET);
+	}
+
+	/**
+	 * @return boolean
+	 */
+	public function isCroppingAllowed() {
+		if ($this->width !== NULL && $this->height !== NULL) {
+			return TRUE;
+		}
+		return $this->allowCropping;
+	}
+
+	/**
+	 * @return boolean
+	 */
+	public function isUpScalingAllowed() {
+		if ($this->width !== NULL && $this->height !== NULL) {
+			return TRUE;
+		}
+		return $this->allowUpScaling;
```

Comment:
UpScaling is always allowed if width and height is set to respect width / height